### ISSUE TITLE
feat: create GitHub releases on `changeset publish`

### ISSUE THREAD

**Issue by JounQin** [changesets/changesets#885]
## Affected Packages

`@changesets/cli`

## Problem

We're publishing to npm manually in https://github.com/mdx-js/eslint-mdx, and `changeset publish` does not create GitHub releases automatically, this functionality is only available in changesets' action

## Proposed solution

https://github.com/changesets/action/blob/6a46d2c28377497e64763ba5f1faeabbd3d88c33/src/run.ts#L94

Move `pushTags` and `createReleases` into `changeset publish`

**Comment by silesky** [comment]
FYI, I had to write this myself recently https://github.com/segmentio/analytics-next/blob/master/scripts/create-release-from-tags/index.ts

**Comment by dotansimha** [comment]
I think at the moment the core of changesets doesn't have anything specific to GitHub? It does some Git work, but it doesn't deal with features like Releases. That's the role of the GitHub Action/Bot?

**Comment by JounQin** [comment]
> I think at the moment the core of changesets doesn't have anything specific to GitHub? It does some Git work, but it doesn't deal with features like Releases. That's the role of the GitHub Action/Bot?

I understand, but for publishing manually, how can a GitHub/Bot help?

**Comment by silesky** [comment]
> I think at the moment the core of changesets doesn't have anything specific to GitHub? It does some Git work, but it doesn't deal with features like Releases. That's the role of the GitHub Action/Bot?

[changelog-github](https://github.com/changesets/changesets/tree/main/packages/changelog-github) is in the repo, so there is  precedent for some github integration here, though it's opt-in, so whether it's considered "core" is debatable.

**Comment by JounQin** [comment]
Or maybe we can do GitHub release on GitHub Action without npm publishing, how do you think?

I recently met this problem again on releasing [`vscode-mdx`](https://github.com/mdx-js/vscode-mdx) which is not a npm package. 😂

**Comment by wu-json** [comment]
Yeah I would prefer that creating GitHub release would be in a separate command like @JounQin proposed. Makes it possible to still automate Git Releases for private repositories that shouldn't be published anywhere.

**Comment by wu-json** [comment]
It would be great if this could be a `changesets` cli command as well.
